packages/devextreme/js/__internal/ui/splitter/utils/splitter.layout_helper.ts

```diff
@@ -0,0 +1,104 @@
+import $ from '@js/core/renderer';
+import {
+  normalizeStyleProp, styleProp,
+} from '@js/core/utils/style';
+
+const ORIENTATION = {
+  horizontal: 'horizontal',
+  vertical: 'vertical',
+};
+
+const FLEX_PROPERTY_NAME = 'flexGrow';
+
+export default class SplitterLayoutHelper {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private readonly items: any;
+
+  private readonly orientation: string;
+
+  private layoutState: number[];
+
+  constructor(items: unknown, orientation: string) {
+    this.items = items;
+    this.layoutState = [];
+    this.orientation = orientation;
+  }
+
+  initializeState(): void {
+    this.layoutState = [];
+
+    this.items.each((index, splitterItem) => {
+      // @ts-expect-error todo: fix error
+      this.layoutState.push(parseFloat($(splitterItem).css(FLEX_PROPERTY_NAME)));
+      //
+    });
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  applyNewLayout(e): void {
+    const newLayoutState = this._getNewLayout(e);
+
+    this.items.each((index, item) => {
+      this._setFlexProp(item, FLEX_PROPERTY_NAME, newLayoutState[index]);
+    });
+  }
+
+  layoutItems(): void {
+    // NOTE: temporarily evenly distributed
+    const splitterItemRatio = 100 / this.items.length;
+
+    this.items.each((index, item) => {
+      this._setFlexProp(item, FLEX_PROPERTY_NAME, splitterItemRatio);
+    });
+  }
+
+  // eslint-disable-next-line max-len
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, class-methods-use-this
+  _setFlexProp(element, prop, value): void {
+    const normalizedProp = normalizeStyleProp(prop, value);
+    element.style[styleProp(prop)] = normalizedProp;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  _getOffset(e): number {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return this.orientation === ORIENTATION.horizontal ? e.offset.x : e.offset.y;
+  }
+
+  _getSplitterItemsSizeSum(): number {
+    const itemSizesSum = this.items.toArray().reduce((total, item) => {
+      const itemRect = item.getBoundingClientRect();
+      const itemSize = this.orientation === ORIENTATION.horizontal
+        ? itemRect.width : itemRect.height;
+
+      // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+      const result = total + itemSize;
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      return result;
+    }, 0);
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return itemSizesSum;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  _getNewLayout(e): number[] {
```

Comment:
There are a lot of mistakes in working of this code. We should test each scenario so that we are assured everything works as expected.